lib/SimpleSAML/Locale/Language.php

```diff
@@ -377,6 +384,16 @@ public function isLanguageRTL(): bool
         return in_array($this->getLanguage(), $this->rtlLanguages, true);
     }
 
+    /**
+     * Returns the list of languages in order of preference. This is useful
+     * to search e.g. an array of entity names for first the current language,
+     * if not present the default language, if not present the fallback language.
+     */
+    public function getPreferredLanguages(): array
+    {
+        $curLanguage = $this->getLanguage();
+        return array_unique([0 => $curLanguage, 1 => $this->defaultLanguage, 2 => self::FALLBACKLANGUAGE]);
```

Comment:
Should we also consider the language priorities from the configuration here?

Reply:
Yes, that would make sense. Even better to have the code in one place so we can consider all available options